metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java

```diff
@@ -169,7 +173,46 @@ public abstract VersionedAspect getVersionedAspect(@Nonnull final Urn urn, @Nonn
   @Nonnull
   protected abstract UpdateAspectResult ingestAspectToLocalDB(@Nonnull final Urn urn, @Nonnull final String aspectName,
       @Nonnull final Function<Optional<RecordTemplate>, RecordTemplate> updateLambda,
-      @Nonnull final AuditStamp auditStamp, @Nonnull final SystemMetadata providedSystemMetadata);
+      @Nonnull final AuditStamp auditStamp, @Nonnull final SystemMetadata systemMetadata);
+
+  /**
+   * Same as ingestAspectToLocalDB but for multiple aspects
+   */
+  @Nonnull
+  protected abstract List<Pair<String, UpdateAspectResult>> ingestAspectsToLocalDB(@Nonnull final Urn urn,
+    @Nonnull List<Pair<String, RecordTemplate>> aspectRecordsToIngest,
+    @Nonnull final AuditStamp auditStamp, @Nonnull final SystemMetadata providedSystemMetadata);
+
+  @Nonnull
+  private SystemMetadata generateSystemMetadataIfEmpty(SystemMetadata systemMetadata) {
+    if (systemMetadata == null) {
+      systemMetadata = new SystemMetadata();
+      systemMetadata.setRunId(DEFAULT_RUN_ID);
+      systemMetadata.setLastObserved(System.currentTimeMillis());
+    }
+    return systemMetadata;
+  }
+
+  private void validateUrn(@Nonnull final Urn urn) {
+    if (!urn.toString().trim().equals(urn.toString())) {
+      throw new IllegalArgumentException("Error: cannot provide an URN with leading or trailing whitespace");
+    }
+  }
+
+  public void ingestAspects(@Nonnull final Urn urn, @Nonnull List<Pair<String, RecordTemplate>> aspectRecordsToIngest,
```

Comment:
We should add a unit test for the ingestAspects method in EbeanEntityServiceTest!